src/modules/member/service/password.service.ts

```diff
@@ -119,4 +135,249 @@ export class PasswordService {
+            {
+              email,
+            },
+          ],
+        })
+        if (!member) {
+          return [member, 'Email is not already used']
+        }
+      } catch (error) {
+        return [member, error]
+      }
+
+      this.logger.info(`Done InquiryMemberExistByEmailFunc ${dayjs().diff(start)} ms`)
+      return [member, '']
+    }
+  }
+
+  async UpdateLoginTokenToMemberFunc(): Promise<UpdateLoginTokenToMemberType> {
+    return async (member: Member, loginToken: string): Promise<string> => {
+      const start = dayjs()
+      try {
+        member.loginToken = loginToken
+        await member.save()
+      } catch (error) {
+        return error
+      }
+
+      this.logger.info(
+        `Done updateLoginTokenToMemberFunc ${dayjs().diff(start)} ms`,
+      )
+      return ''
+    }
+  }
+
+  ResetPasswordEmailHandler(
+    inquiryMemberExistByLoginTokenAndEmail: Promise<InquiryMemberExistByLoginTokenAndEmailType>,
+    updateLoginTokenToMember: Promise<UpdateLoginTokenToMemberType>,
+    updatePasswordToMember: Promise<UpdatePasswordToMemberType>,
+  ) {
+    return async (body: ResetPasswordEmailRequestDto) => {
+      const start = dayjs()
+
+        const { loginToken, email } = body
+
+        const [member, inquiryMemberExistByLoginTokenAndEmailError] = await (await inquiryMemberExistByLoginTokenAndEmail)(
+          loginToken,
+          email
+        )
+
+        if (inquiryMemberExistByLoginTokenAndEmailError !== '') {
+          return response(
+            undefined,
+            UnableInquiryMemberExistByLoginTokenAndEmailError,
+            inquiryMemberExistByLoginTokenAndEmailError,
+          )
+        }
+
+      const updatePasswordToMemberError = await (await updatePasswordToMember)(
+        member,
+        body.password,
+      )
+      if (updatePasswordToMemberError !== '') {
+        return internalSeverError(
+          UnableUpatePasswordToDb,
+          updatePasswordToMemberError,
+        )
+      }
+
+      const updateLoginTokenToMemberError = await (await updateLoginTokenToMember)(
+        member,
+        null,
+      )
+      if (updateLoginTokenToMemberError !== '') {
+        return internalSeverError(
+          UnableUpateLoginTokenToDb,
+          updateLoginTokenToMemberError,
+        )
+      }
+
+      this.logger.info(`Done forgotPasswordHandler ${dayjs().diff(start)} ms`)
+      return response(undefined)
+    }
+  }
+
+  async InquiryMemberExistByLoginTokenAndEmailFunc(
+    etm: EntityManager,
+  ): Promise<InquiryMemberExistByLoginTokenAndEmailType> {
+    return async (loginToken: string, email: string): Promise<[Member, string]> => {
+      const start = dayjs()
+      let member: Member
+      try {
+        member = await etm.findOne(Member, {
+          where: [
+            {
+              loginToken,
+              email,
+            },
+          ],
+        })
+        if (!member) {
+          return [member, 'Can not reset password']
```

Comment:
แก้เป็น Can't reset password beacase member not found. ใส่เหตุผลไปด้วย บอกแค่ว่า Can not reset password ดูจาก response code ก็รู้ว่าทำไม่ได้